### PR TITLE
Log errors only when connected to diamond socket

### PR DESCRIPTION
**Pull request by gnufied**

### CODE REVIEW

File: src/diamond/collector.py [hunk 1]
```diff
@@ -131,6 +134,11 @@ def load_config(self, config):
                     self.config.update(config['diamondCollectors'][self.name])
         self.process_config()
 
+    def can_publish_metric(self):
+        if ((self._socket is not None) and (self._reconnect is False)):
```
Comment (venkey-ariv) [inline src/diamond/collector.py]: return (self._socket is not None) and (self._reconnect is False)

**Comment by zachm** [comment]
remove parens on :138. otherwise, +1

**Comment by venkey-ariv** [comment]
lgtm (apart from what what ztm has mentioned)

**Comment by tsheasha** [comment]
+1

**Pull request merged**